### PR TITLE
Fix for the problem in the `NestedRows` plugin

### PR DESCRIPTION
**Pull request by wszymanski**
### Context
Fix for the bug. Replaced the `modifyRemovedAmount` with `modifyRemovedRows` hook.

### How has this been tested?

Just the issue from the case was tested. All `e2e` tests pass.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/6890

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/dataMap.js [hunk 1]
```diff
@@ -413,21 +413,26 @@ class DataMap {
    * @returns {boolean} Returns `false` when action was cancelled, otherwise `true`.
    */
   removeRow(index, amount = 1, source) {
-    let rowIndex = typeof index !== 'number' ? -amount : index;
-    const rowsAmount = this.instance.runHooks('modifyRemovedAmount', amount, rowIndex);
+    let rowIndex = Number.isInteger(index) ? index : -amount; // -amount = taking indexes from the end.
+    const physicalIndexes = this.visualRowsToPhysical(rowIndex, amount);
+    // It handle also callback from the `NestedRows` plugin. Removing parent node has effect in removing children nodes.
+    const removedPhysicalIndexes = this.instance.runHooks('modifyRemovedRows', physicalIndexes);
```
Comment (budnix) [inline src/dataMap.js:419]: it's worth to add a hook to documentation (`pluginHooks.js`).

Comment (wszymanski) [reply src/dataMap.js:419]: @wojciechczerniak, the old `modifyRemovedAmount` was quite private. New, `modifyRemovedRows` seems to be more useful.

There is even a test for extra case: https://github.com/handsontable/handsontable/pull/6907/commits/cd3b38c646ece46ee84ed66b4d1e6e93bfb448a4, so for me, the situation isn't similar to this introduced within [PR discussion](https://github.com/handsontable/handsontable/pull/6889#discussion_r426590991). I would add the hook and describe it in the release notes. WDYT?

Comment (wszymanski) [reply src/dataMap.js:419]: I suppose that it should be added after watching the mentioned discussion. Changed within: https://github.com/handsontable/handsontable/pull/6907/commits/650aa1824a666cea705184c5e4fe1d62c96349fa.

Comment (wszymanski) [reply src/dataMap.js:419]: @aninde Please add information to the release notes :)

Comment (wszymanski) [reply src/dataMap.js:419]: @scarletfog Please add information to the release notes :)

Comment (scarletfog) [reply src/dataMap.js:419]: @wszymanski can we say it's a new feature?

Comment (wszymanski) [reply src/dataMap.js:419]: I think so.

Comment (scarletfog) [reply src/dataMap.js:419]: ok, thank you! 😄

**Comment by wszymanski** [comment]
@scarletfog Finally, after applying @wojciechczerniak suggestion we use old, known hook `beforeRemoveRow`.

**Pull request merged**